extractor/src/main/java/org/schabi/newpipe/extractor/services/bandcamp/extractors/BandcampRadioStreamExtractor.java

```diff
@@ -0,0 +1,142 @@
+        try {
+            return JsonParser.object().from(
+                    NewPipe.getDownloader().get("https://bandcamp.com/api/bcweekly/1/get?id=" + id).responseBody()
+            );
+        } catch (final IOException | ReCaptchaException | JsonParserException e) {
+            throw new ParsingException("could not get show data", e);
+        }
+    }
+
+    @Override
+    public void onFetchPage(@Nonnull final Downloader downloader) throws IOException, ExtractionException {
+        showInfo = query(Integer.parseInt(getId()));
+    }
+
+    @Nonnull
+    @Override
+    public String getName() throws ParsingException {
+        return showInfo.getString("subtitle"); // "audio_title" is a boring title
+    }
+
+    @Nonnull
+    @Override
+    public String getUploaderUrl() throws ContentNotSupportedException {
+        throw new ContentNotSupportedException("Fan pages are not supported");
+    }
+
+    @Nonnull
+    @Override
+    public String getUrl() throws ParsingException {
+        return getLinkHandler().getUrl();
+    }
+
+    @Nonnull
+    @Override
+    public String getUploaderName() {
+        return Jsoup.parse(showInfo.getString("image_caption"))
+                .getElementsByTag("a").first().text();
+    }
+
+    @Nullable
+    @Override
+    public String getTextualUploadDate() {
+        return showInfo.getString("published_date");
+    }
+
+    @Nonnull
+    @Override
+    public String getThumbnailUrl() throws ParsingException {
+        return getImageUrl(showInfo.getLong("show_image_id"), false);
+    }
+
+    @Nonnull
+    @Override
+    public String getUploaderAvatarUrl() {
+        return "https://bandcamp.com/img/buttons/bandcamp-button-circle-whitecolor-512.png";
+    }
+
+    @Nonnull
+    @Override
+    public Description getDescription() {
+        return new Description(showInfo.getString("desc"), Description.PLAIN_TEXT);
+    }
+
+    @Override
+    public long getLength() {
+        return showInfo.getLong("audio_duration");
+    }
+
+    @Override
+    public List<AudioStream> getAudioStreams() {
+        final ArrayList<AudioStream> list = new ArrayList<>();
+        final JsonObject streams = showInfo.getObject("audio_stream");
+
+        if (streams.has("opus-lo")) {
+            list.add(new AudioStream(
+                    streams.getString("opus-lo"),
+                    MediaFormat.OPUS, 100
+            ));
+        }
+        if (streams.has("mp3-128")) {
+            list.add(new AudioStream(
+                    streams.getString("mp3-128"),
+                    MediaFormat.MP3, 128
+            ));
+        }
+
+        return list;
+    }
+
+    @Nonnull
+    @Override
+    public String getLicence() {
+        return "";
+    }
+
+    @Nonnull
+    @Override
+    public String getCategory() {
+        return "";
+    }
+
+    @Nonnull
+    @Override
+    public List<String> getTags() {
+        return new ArrayList<>();
+    }
```

Comment:
I found 3 ways with which you return an empty list. chose number 1 or 3

1.  `Collections.emptyList()`
2.  `new ArrayList<>()`
3. `Collections.EMPTY_LIST`

Reply:
The correct one is the first one